pkg/provider/provider.go

```diff
@@ -33,28 +40,68 @@ func Provider() *schema.Provider {
 				Optional:      true,
 				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PASSWORD", nil),
 				Sensitive:     true,
-				ConflictsWith: []string{"browser_auth", "private_key_path", "oauth_access_token"},
+				ConflictsWith: []string{"browser_auth", "private_key_path", "oauth_access_token", "oauth_refresh_token"},
 			},
 			"oauth_access_token": {
 				Type:          schema.TypeString,
 				Optional:      true,
 				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_OAUTH_ACCESS_TOKEN", nil),
 				Sensitive:     true,
-				ConflictsWith: []string{"browser_auth", "private_key_path", "password"},
+				ConflictsWith: []string{"browser_auth", "private_key_path", "password", "oauth_refresh_token"},
+			},
+			"oauth_refresh_token": {
```

Comment:
We should probably update the documentation to reflect these new environment variables and authentication mechanisms.

Reply:
working on documentation

Reply:
updated documentation `~/docs/index.md` to reflect OAuth through Refresh token details, such as setting them as environment variable and authentication mechanisms. Please review and advice